tokio-util/src/codec/decoder.rs

```diff
@@ -159,7 +158,7 @@ pub trait Decoder {
     /// [`Framed`]: crate::codec::Framed
     fn framed<T: AsyncRead + AsyncWrite + Sized>(self, io: T) -> Framed<T, Self>
     where
-        Self: Encoder + Sized,
+        Self: Sized,
```

Comment:
Why was this bound removed?

Reply:
Because it can't be written now without causing type inference failures. Adding a type parameter `I` for `Encoder<I>` would not work since it does not appear in the returned `Framed` type. https://github.com/tokio-rs/tokio/pull/1746#issuecomment-576600664 .

Reply:
Ah, we should probably change decoder to also take a generic type as well then.

Reply:
Nope, that won't work. There is no reason that the types should be the same as they just represent what is read or written to the underlying IO stream (http request/response) etc.